CONTRIBUTING.md

```diff
@@ -0,0 +1,643 @@
+   If it is implemented with anything else (e.g. TensorFlow), we cannot add it to Spandrel. The repos of the architecture should have a `requirements.txt` file that lists all dependencies. If it lists `torch` as a dependency, we are good to go.
+2. Where we can get official models. \
+   Pretty much all architectures have official models trained by the researchers that made the architecture. Typically, the README of the repo will give you links to these models (often Google Drive). Some also use GitHub releases. Take note of these links for now. We will later use the official models in our tests.
+3. The license of the architecture. \
+   If the architecture's code is not licensed under a permissive license (e.g. MIT, BSD, Apache), we cannot add it to Spandrel. If there is no license or the license is not clear, you have to ask the authors for permission to copy their code. A permissive license or explicit permission from the authors is required for adding an architecture to Spandrel.
+
+In the case of DITN, the architecture is implemented using PyTorch, a link to the official models is in the README, and it is licensed.
+
+### Step 1: Copy the architecture code
+
+The first step is to copy the code of the architecture into Spandrel.
+
+Of course, the copy this code, we first have to find it! Unfortunately, project layout is not consistent across architecture repos, so you might be to search a bit. You need that file with the class that defines the architecture. In most repos, this file is called `model.py`, `models/<arch name>.py`, or `basicsr/archs/<arch name>_arch.py`. You know that you found the right file when it contains a class that is named after the architecture and inherits from `nn.Module`.
+
+In the case of DITN, the file is called [`models/DITN_Real.py`](https://github.com/yongliuy/DITN/blob/3438e429c0538ee5061a7cfca587df0c4097703f/models/DITN_Real.py#L197).
+
+Once you have found the file, create a new directory `src/spandrel/architectures/<arch name>/arch/` and copy the file into it. This directory will contain all code that we will copy. Since we respect the copy right of the original authors, we will also copy the `LICENSE` file of the repo into this directory.
+
+In the case of DITN, we copy `models/DITN_Real.py` to `src/spandrel/architectures/DITN/arch/DITN_Real.py` and add the `LICENSE` file of the repo.
+
+The main model file might also reference other files in the repo. You have to copy those 2 into your `arch/` directory as well.
+
+In the case of DITN, the main model file doesn't reference any other files, so we are done.
+
+### Step 1.1: Linting, type errors, and cleaning up
+
+You might have already noticed that the code you copied lights up like a Christmas tree in your IDE. In VSCode, just hit Ctrl+S to let ruff fix the formatting and most linting errors.
+
+However, there are a few things that we have to fix manually.
+
+- "Use `super()` instead of `super(__class__, self)`" \
```

Comment:
Isn't there a ruff rule we can enable to fix this automatically?

Reply:
The rule doesn't auto fix, it only offers an IDE suggestion. That's why I mentioned this.